src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -96,12 +107,32 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
 
         await this._paypalCommercePaymentProcessor.initialize(paramsScript, undefined, gatewayId);
 
+        const fundingKey = this._paypalCommerceFundingKeyResolver.resolve(methodId, gatewayId);
+
         this._paypalCommercePaymentProcessor.renderButtons(cartId, container, buttonParams, {
             onRenderButton,
-            fundingKey: this._paypalCommerceFundingKeyResolver.resolve(methodId, gatewayId),
+            fundingKey,
             paramsForProvider: {isCheckout: true},
         });
 
+        if (this._isAPM)  {
+            /**
+             * FALLBACK for old checkouts that didn't pass apmFieldsContainer id
+             */
+            const apmFieldsContainerId = apmFieldsContainer || `${container}apmFieldsContainer`;
+            const fullName = `${firstName} ${lastName}`;
+
+            if (!document.querySelector(apmFieldsContainerId)) {
+                const apmFieldsContainerElement = document.createElement('div');
+                apmFieldsContainerElement.setAttribute('id', apmFieldsContainerId.substring(1));
+
+                const buttonContainer = document.querySelector(container);
+                buttonContainer?.insertBefore(apmFieldsContainerElement, buttonContainer.firstChild);
```

Comment:
I think we should go with the existing approach (when the payment widget is next to the radio button) https://github.com/bigcommerce/checkout-js/blob/051363765fb15c03d818dcb09d0668be574f490a/src/app/payment/paymentMethod/HostedWidgetPaymentMethod.tsx#L269, rather than inject widget before the button container

Reply:
what if `buttonContainer` is empty? Should we be throwing an error? Otherwise, it might just fail silently and we won't have any visibility.

Reply:
To be 100% sure, I’ll add check if there is container in DOM on initialization of PAYPAL. To avoid problem with button and fields initialization.